Documentation/policies/package-servicing.md

```diff
@@ -0,0 +1,50 @@
+# .NET Package Servicing Model
+
+Most packages in the .NET ecosystem support only the latest version. Many packages from the .NET team operate on this same model. A subset of our packages operate on another model where multiple version bands are supported at the same time, matching in-support runtimes, offering more flexibility to users.
+
+Users should aim to maintain their package references to be as close to the supported version as possible. Being close to a supported version means that you have most or all security fixes and likely have no latent compatibility risk that can be present when there is a big gap between your reference and the supported version.
+
+## Latest version servicing
+
+The default package servicing model is “latest version”. The team publishes an updated package with a security and/or functionality fix. Users can upgrade to the new latest version to get the fix.
+
+The following libraries use this servicing model:
+
+- [.NET Aspire](https://github.com/dotnet/aspire)
+- [ML.NET](https://github.com/dotnet/machinelearning)
+- [Maintenance-packages](https://github.com/dotnet/maintenance-packages)
+
+## Runtime-band servicing
+
+Some packages match a library (in name and content) in the runtime. In this case, the team publishes  fixes for the library in the runtime and the package at the same time. The version numbers will match. Users can upgrade to the new latest patch version to get the fix.
+
+Users can upgrade to the latest library version (like 8.0.x) that matches a supported runtime (like 8.0). Users can also move to the latest patch of a newer major version (like 9.0.x) provided that the runtime is still in support.
+
+The advantage of matching versions is that the runtime library will be used in place of the package, resulting in smaller deployments and servicing being provided by the runtime. The advantage of using a newer package version is getting access to newer functionality.
+
+Apps only ever need a package reference to one of these libraries to use a newer major version. It is more common for libraries to have a reference, particularly .NET Standard libraries.
+
+The following libraries use this servicing model:

```

Comment:
Should we add EF Core here?

cc @AndriySvyryd @SamMonoRT

Reply:
The lists are not intended to be exhaustive. However, 3-6 examples would be great. Feel free to add in more via suggestion. It seems like an exhaustive list should be provided via a different approach.

Reply:
One idea we had last week in a meeting in this area was:

1. Mark old package versions as "deprecated" on nuget.org. (for example, all System.Text.Json 7.0.x versions would be marked as deprecated today.)
2. Add tooling to let you know when you are using something is not supported any longer.

This way, you could see that the 8.x and 9.x versions were still in support. But the 7.x version isn't.

For "Latest version servicing" like .NET Aspire all 8.x versions would be marked as "deprecated".

Reply:
> exhaustive list 

I'm not suggesting this list be exhaustive of all packages. What I am suggesting is to document the _groups_ of packages that follow the same model. All Runtime, ASP.NET Core, EF Core follow this model. All Extensions, Aspire, ML.NET follow latest version model.

Reply:
I would love for support level to be exposed via the NuGet gallery and the NuGet API/client. That's clearly the best model.

@OliaG @aortiz-msft

Reply:
> The advantage of matching versions is that the runtime library will be used in place of the package, resulting in smaller deployments and servicing being provided by the runtime.

This does not apply to EF, perhaps there should be a subsection for packages in the shared framework.

Reply:
This doc cannot become a complex map of all the packages and how they are different.  It needs to describe some basic principles that folks can understand and get value from.

Reply:
Sure. But right now it's factually incorrect.

Reply:
Feel free to write a doc for EF (in the EF repo) and we can link to it.

Reply:
> I would love for support level to be exposed via the NuGet gallery and the NuGet API/client. That's clearly the best model. That's a good idea to follow merging this PR.
> 
> @OliaG @aortiz-msft

@richlander Long thread here. Would you please summarize the ask for us and open an issue in NuGet/Home? 

cc @OliaG

Reply:
Ask: There should be an E2E for deprecated/unsupported packages from producer marking existing packages as such and consumers getting warnings via the gallery, VS, and the NuGet/dotnet CLI. A team like ours isn't going to use the gallery to mark our packages as deprecated. We need an API for that (we're talking to @joelverhagen about that part). I can open an issue if there isn't one on that.

Reply:
https://github.com/NuGet/Home/pull/13598 ?

Reply:
Yes. That appears to be a match and I remember it now. Good find.

FYI: @JonDouglas